pkg/generate/generate.go

```diff
@@ -0,0 +1,138 @@
+import "os"
+import "path"
+import "text/template"
+import "bytes"
+
+const componentBuilder = `apiVersion: bundle.gke.io/v1alpha1
+kind: ComponentBuilder
+componentName: etcd-component
+version: 1.0.0
+objectFiles:
+- url: /sample-deployment.yaml
+- url: /sample-service.yaml
+- url: /sample-patch.yaml
+`
+const sampleDeployment = `apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: {{.Name}}
+  name: helloworld
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{.Name}}
+      name: {{.Name}}
+    spec:
+      containers:
+        - name: {{.Name}}
+          image: gcr.io/google-samples/hello-app:2.0
+`
+const sampleService = `apiVersion: v1
+kind: Service
+metadata:
+  name: {{.Name}}
+  labels: 
+    app: {{.Name}}
+spec:
+  type: NodePort
+  ports:
+  - name: {{.Name}}
+    port: 8080
+    targetPort: 8080
+  selector:
+    app: {{.Name}}
+`
+
+const patchTemplate = `apiVersion: bundle.gke.io/v1alpha1
+kind: PatchTemplate
+template: |
+  apiVersion: apps/v1beta1
+  kind: Deployment
+  metadata:
+    namespace: {{.namespace}}
+optionsSchema:
+  required:
+  - namespace
+  properties:
+    namespace:
+      type: string
+      pattern: '^[a-z0-9-]+$'
+`
+const patchOptions = `# Options for applying to the component via patch-templates
+namespace: foo-namespace
+buildLabel: test-build
+`
+
+// Create scaffolds basic set of files to the filesystem
+func Create(filepath string, name string) error {
+	var writeErr error
+	if err := os.Mkdir(filepath, 0777); err != nil {
+		writeErr = err
+		return writeErr
+	}
+
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-component-builder.yaml"), []byte(componentBuilder), 0666); err != nil {
+		writeErr = err
+	}
+	replacement := struct {
+		Name string
+	}{
+		Name: name,
+	}
+
+	deploymentTemplate, _ := template.New("deployment").Parse(sampleDeployment)
+	var deploymentText bytes.Buffer
+	deploymentTemplate.Execute(&deploymentText, replacement)
+
+	serviceTemplate, _ := template.New("service").Parse(sampleService)
+	var serviceText bytes.Buffer
+	serviceTemplate.Execute(&serviceText, replacement)
+
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-deployment.yaml"), deploymentText.Bytes(), 0666); err != nil {
+		writeErr = err
+		goto handleError
+	}
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-service.yaml"), serviceText.Bytes(), 0666); err != nil {
+		writeErr = err
+		goto handleError
+	}
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-patch.yaml"), []byte(patchTemplate), 0666); err != nil {
+		writeErr = err
+		goto handleError
+	}
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-patch.yaml"), []byte(patchTemplate), 0666); err != nil {
+		writeErr = err
+		goto handleError
+	}
+	if err := ioutil.WriteFile(path.Join(filepath, "sample-options.yaml"), []byte(patchOptions), 0666); err != nil {
+		writeErr = err
+		goto handleError
+	}
+
+handleError:
+	if writeErr != nil {
+		os.Remove(filepath)
```

Comment:
I think it's fine to write partial results. Also, the go-to stuff is pretty weird

Reply:
okay.  I think in practice that should work well.